[Cocoa] Youtube videos don't always size correctly, both inline and fullscreen

jernoble opened the pull request:
#### 8addb776de1b2069f92a44ec9571bc6afaa51221
<pre>
[Cocoa] Youtube videos don&apos;t always size correctly, both inline and fullscreen
<a href="https://bugs.webkit.org/show_bug.cgi?id=255824">https://bugs.webkit.org/show_bug.cgi?id=255824</a>
rdar://106974872

Reviewed by Simon Fraser.

When this behavior occurs, the video layer fails to resize itself when layout changes. E.g.,
when going into fullscreen mode, the video layer will retain the original size it had inline.
(Which, to be clear, may also be incorrect.) The layer tree indicates that the incorrect size
is based on a 2:1 (or 300x150) aspect ratio, and the failure to resize indicates that the
VideoFullscreenInterfaceAVKit thinks its videoDimensions are 0x0.

This prospective fix seeks to eliminate a timing issue where, perhaps, the &quot;resize&quot; event is
not being fired when the underlying MediaPlayer changes size. It also adds more logging around
resizing in case this issue happens again that will hopefully aid in diagnostics.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setReadyState):
(WebCore::HTMLMediaElement::mediaPlayerSizeChanged):
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::scheduleResizeEvent):
(WebCore::HTMLMediaElement::scheduleResizeEventIfSizeChanged):
* Source/WebCore/html/HTMLVideoElement.cpp:
(WebCore::HTMLVideoElement::scheduleResizeEvent):
(WebCore::HTMLVideoElement::scheduleResizeEventIfSizeChanged):
* Source/WebCore/html/HTMLVideoElement.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::updateCachedState):

Canonical link: <a href="https://commits.webkit.org/263292@main">https://commits.webkit.org/263292@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef34ad724bd3a1028bf6285aacbdd65bade62339

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4125 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4236 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4354 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5586 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4368 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4113 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4334 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4201 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/4595 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4182 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4361 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3723 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5579 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1858 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3700 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/5872 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3685 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3763 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5270 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4160 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3370 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3674 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3698 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1021 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/7807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3953 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/html/HTMLVideoElement.h:
```diff
@@ -148,8 +148,7 @@ class HTMLVideoElement final : public HTMLMediaElement, public Supplementable<HT
 
-    unsigned m_lastReportedVideoWidth { 0 };
-    unsigned m_lastReportedVideoHeight { 0 };
+    FloatSize m_lastReportedNaturalSize { };
```
No need for `{ }`

smfr reviewed (approved):
Can we write a test for this?

jernoble committed:
[Cocoa] Youtube videos don't always size correctly, both inline and fullscreen
https://bugs.webkit.org/show_bug.cgi?id=255824
rdar://106974872

Reviewed by Simon Fraser.

When this behavior occurs, the video layer fails to resize itself when layout changes. E.g.,
when going into fullscreen mode, the video layer will retain the original size it had inline.
(Which, to be clear, may also be incorrect.) The layer tree indicates that the incorrect size
is based on a 2:1 (or 300x150) aspect ratio, and the failure to resize indicates that the
VideoFullscreenInterfaceAVKit thinks its videoDimensions are 0x0.

This prospective fix seeks to eliminate a timing issue where, perhaps, the "resize" event is
not being fired when the underlying MediaPlayer changes size. It also adds more logging around
resizing in case this issue happens again that will hopefully aid in diagnostics.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setReadyState):
(WebCore::HTMLMediaElement::mediaPlayerSizeChanged):
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::scheduleResizeEvent):
(WebCore::HTMLMediaElement::scheduleResizeEventIfSizeChanged):
* Source/WebCore/html/HTMLVideoElement.cpp:
(WebCore::HTMLVideoElement::scheduleResizeEvent):
(WebCore::HTMLVideoElement::scheduleResizeEventIfSizeChanged):
* Source/WebCore/html/HTMLVideoElement.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::updateCachedState):

Canonical link: https://commits.webkit.org/263292@main
